// data-analysis-python\src\get_jp_tickers\get_jp_tickers.proto
syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/ms_gateway/get_jp_tickers";

// 日本株ティッカーを取得するためのサービスを定義
service GetJpTickersService {
  rpc GetJpTickers (GetJpTickersRequest) returns (GetJpTickersResponse);
}

// リクエストメッセージ（パラメータが不要の場合でも定義）
message GetJpTickersRequest {}

// 日本株ティッカーを格納するレスポンスメッセージ
message GetJpTickersResponse {
  repeated string tickers = 1;  // ティッカーシンボルのリスト
}

// python用コンパイルコマンド /data-analysis-pythonにて
// python -m grpc_tools.protoc -I src/get_jp_tickers --python_out=src/get_jp_tickers --grpc_python_out=src/get_jp_tickers src/get_jp_tickers/get_jp_tickers.proto

// Go用コンパイルコマンド /api-goにて
// protoc --proto_path=../data-analysis-python/src/get_jp_tickers --go_out=./src/service/ms_gateway/get_jp_tickers --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/get_jp_tickers --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_jp_tickers/get_jp_tickers.proto
